L1Trigger/L1TGlobal/plugins/TriggerMenuParser.cc

```diff
@@ -172,7 +182,30 @@ void l1t::TriggerMenuParser::setGtAlgorithmAliasMap(const AlgorithmMap& algoMap)
 
 std::map<std::string, unsigned int> l1t::TriggerMenuParser::getExternalSignals(const L1TUtmTriggerMenu* utmMenu) {
   using namespace tmeventsetup;
-  const esTriggerMenu* menu = reinterpret_cast<const esTriggerMenu*>(utmMenu);
+  const std::map<std::string, L1TUtmCondition>& condMap = utmMenu->getConditionMap();
+
+  std::map<std::string, unsigned int> extBitMap;
+
+  //loop over the algorithms
+  for (std::map<std::string, L1TUtmCondition>::const_iterator cit = condMap.begin(); cit != condMap.end(); cit++) {
+    const L1TUtmCondition& condition = cit->second;
+    if (condition.getType() == esConditionType::Externals) {
+      // Get object for External conditions
+      const std::vector<L1TUtmObject>& objects = condition.getObjects();
+      for (size_t jj = 0; jj < objects.size(); jj++) {
```

Comment:
same comment about range based loop

Reply:
This will be adjusted in the next update